toolchains/solidity/extension/src/tests/test-manager.ts

```diff
@@ -0,0 +1,280 @@
+import { LanguageClient } from 'vscode-languageclient/node';
+import * as vscode from 'vscode';
+import { testContract, testFunction, FileResult } from './foundry-test';
+
+enum ItemType {
+    file,
+    contractCase,
+    testCase
+}
+
+export class TestManager {
+    public testController: vscode.TestController;
+    private testData = new WeakMap<vscode.TestItem, ItemType>();
+
+    constructor(private client: LanguageClient, private workspace: string) {        
+        this.testController = vscode.tests.createTestController("solidityTestController", "Solidity test controller");
+
+        this.testController.resolveHandler = (test) =>  {
+          console.log("controller resolve");
+          return this.resolve(test);
+        };
+        this.testController.createRunProfile("Run tests", vscode.TestRunProfileKind.Run, (request, token) => this.runHandler(false, request, token));
+        // Uncomment this when debugging is supported
+        //this.testController.createRunProfile("Debug tests", vscode.TestRunProfileKind.Run, (request, token) => this.runHandler(true, request, token))
```

Comment:
comment left either add details or delete, considering the coment above i guess that was from a template